common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java

```diff
@@ -0,0 +1,113 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * A utility which collects any event's information given by calling markCode on such event(s).
+ * These events can be retrieved in csv format by calling method getFileContent.
+ * The event is recognized by a string marker which is prefixed with scenario code.
+ * MAX_SIZE_CODE_MARKER is the maximum number of markers this utility can have.
+ */
+public class CodeMarkerManager {
+
+    private boolean enableCodeMarker = false;
+    private int MAX_SIZE_CODE_MARKER = 1000;
```

Comment:
This probably should be `static`

Reply:
as well as `final`

Reply:
Hi @shahzaibj,
As we are now using the singleton pattern so I feel, having it non-static will give us advantage that this variable will be initialized only when the class CodeMarkerManager.java is used and object is created otherwise not. 

For your suggestion of using "final", I agree, I'm just pushing in the next iteration.

Please share your thoughts. Thanks.

Reply:
Done. Made this static as well.